server/src/domain/media/media.service.ts

```diff
@@ -292,4 +329,10 @@ export class MediaService {
 
     return handler;
   }
+
+  getPath(asset: AssetEntity, extension: string): string {
```

Comment:
The name of this function is quite misleading. It's changing the file system, not just getting a path or something.

Reply:
How about `ensurePath`?

Reply:
Is the mkdir really required? I don't see it used before.

Reply:
It was there before; I just refactored it into a separate method. I'm not sure if the folder is guaranteed to exist prior to the job.

Reply:
We can definitely clean all those up, probably in another PR though. At the moment if the directory doesn't exist the write file will fail.